pandas/core/arrays/period.py

```diff
@@ -241,18 +233,31 @@ def _generate_range(cls, start, end, periods, freq, fields):
 
         return subarr, freq
 
+    def _unbox_scalar(self, value):
+        # type: (Union[Period, NaTType]) -> int
+        if value is NaT:
+            return value.value
+        elif isinstance(value, self._scalar_type):
+            return value.ordinal
+        else:
+            raise ValueError("'value' should be a Period")
+
+    def _scalar_from_string(self, value):
```

Comment:
is this defined in the top-level EA interface?

Reply:
Nope. It may be worth adding https://github.com/pandas-dev/pandas/issues/24261